bot.js

```diff
@@ -172,19 +178,23 @@ function formatDifferenceMessage(difference){
 
 function balance(pool, message){
 
-		/**
-		TODO:
-		Reintroduce Bishop logic
-		Allow leaders to be promoted/demoted and rebalance groups
-		**/
-
-		let [joining, weakest, total] = [pool.slice().sort((a,b) => a.rank-b.rank), [{rank: 100}], [...leaders, ...pool].length]
+		let [bishops, joining, weakest, total] = [pool.filter(member => member.role === "bishop").sort((a,b) => a.rank-b.rank), pool.slice().sort((a,b) => a.rank-b.rank), [{rank: 100}], [...leaders, ...pool].length]
```

Comment:
Should leaders and pool be deduped when accounting for total?

Reply:
What do you mean?